packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -234,7 +238,7 @@ export class FirestoreRepository implements Repository {
   async getPackageVersion(
     packageName: string,
     version: string
-  ): Promise<PackageVersion | undefined> {
+  ): Promise<Omit<PackageVersion, 'customElements' | 'problems'> | undefined> {
```

Comment:
Did the type system catch that you needed to add this Omit, or did you manually have to notice that?

Reply:
I did that because these fields map to sub collections and so aren't part of the PackageVersion document. There's more work to do here to make the mapping from GraphQL types to Firestore types more regularized. I had a utility type to do some of the mapping, but it required some manual configuration like listing the collections. I tried to see if graphql-codegen could adjust types based on field directives (like `@collection` and `@map` that I have in the schema) but it can't do that (yet).

So I'm thinking we should be more formalized here. We don't have _that_ many types in play and we can make a module with utility type like that, that also takes care of specifying the additional denormalized fields, and applies it to all the GraphQL schema types. Then we'll have names for all of these and code/documentation on the specific mapping between types.